From 7683601af9e162e11970e073b15a4e3657caafdd Mon Sep 17 00:00:00 2001
From: Christophe Kerello <christophe.kerello@foss.st.com>
Date: Mon, 20 Jan 2025 14:22:43 +0100
Subject: [PATCH 787/954] feat(st-spi): remove memory mapped support on M33TD

This is a workaround. It will be removed when RISAF sub region
management will be implemented.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git e74ce8bc9b348b62bdb3734ddc0472ac4676572d feat(st-spi): remove memory mapped support on M33TD]
Change-Id: Id6aadfdde97f82dd18f08c723d69e58e15c70670
Signed-off-by: Christophe Kerello <christophe.kerello@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/430034
Reviewed-by: Maxime MERE <maxime.mere@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Domain-Review: Christophe KERELLO <christophe.kerello@st.com>
Reviewed-by: Christophe KERELLO <christophe.kerello@st.com>
Tested-by: Christophe KERELLO <christophe.kerello@st.com>
---
 drivers/st/spi/stm32_ospi.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/st/spi/stm32_ospi.c b/drivers/st/spi/stm32_ospi.c
index cbb84aad33..6d70938bd4 100644
--- a/drivers/st/spi/stm32_ospi.c
+++ b/drivers/st/spi/stm32_ospi.c
@@ -1015,7 +1015,9 @@ static int stm32_ospi_exec_op(const struct spi_mem_op *op)
 
 static int stm32_ospi_dirmap_read(const struct spi_mem_op *op)
 {
+#if defined(STM32MP_M33_TDCID) && !STM32MP_M33_TDCID
 	size_t addr_max;
+#endif /* STM32MP_M33_TDCID */
 	uint8_t fmode = _OSPI_CR_FMODE_INDR;
 
 	if (op->cmd.dtr) {
@@ -1026,10 +1028,12 @@ static int stm32_ospi_dirmap_read(const struct spi_mem_op *op)
 		}
 	}
 
+#if defined(STM32MP_M33_TDCID) && !STM32MP_M33_TDCID
 	addr_max = op->addr.val + op->data.nbytes + 1U;
 	if ((addr_max < stm32_ospi.mm_size) && (op->addr.buswidth != 0U)) {
 		fmode = _OSPI_CR_FMODE_MM;
 	}
+#endif /* STM32MP_M33_TDCID */
 
 	return stm32_ospi_send(op, fmode);
 }
-- 
2.53.0

